src/ngx_http_lua_socket_tcp.c

```diff
@@ -4368,6 +4377,71 @@ ngx_http_lua_socket_tcp_getreusedtimes(lua_State *L)
 
 
 static int
+ngx_http_lua_socket_tcp_getbounddata(lua_State *L)
+{
+    ngx_http_lua_socket_tcp_upstream_t    *u;
+
+    if (lua_gettop(L) != 2) {
+        return luaL_error(L, "expecting 2 argument "
+                          "(including the object), but got %d", lua_gettop(L));
+    }
+
+    luaL_checktype(L, 1, LUA_TTABLE);
+    luaL_checktype(L, 2, LUA_TSTRING);
+
+    lua_rawgeti(L, 1, SOCKET_CTX_INDEX);
+    u = lua_touserdata(L, -1);
+    lua_rawgeti(L, LUA_REGISTRYINDEX, u->bind_tbl_ref);
+    switch (lua_type(L, -1)) {
+            case LUA_TTABLE:
```

Comment:
Style: the switch statement has coding style issues. Please see how we format the switch statement in the nginx code base.